File: android/src/main/kotlin/cn/shper/flutter/tk_logger/TKLoggerFlutter.kt
package cn.shper.flutter.tk_logger

import cn.shper.tklogger.TKLogger
import cn.shper.tklogger.destination.TKLogBaseDestination
import cn.shper.tklogger.model.TKLogModel
import cn.shper.tklogger.thread.ThreadPoolUtils

/**
 * Author : shixupan
 * EMail : shixupan10829@hellobike.com
 * Date : 2020/7/22
 */
object TKLoggerFlutter {

  private val destinations = HashSet<TKLogBaseDestination>()

  /** Destination */
  fun addDestination(destination: TKLogBaseDestination): Boolean {
    if (destinations.contains(destination)) {
      return false
    }

    destinations.add(destination)
    return true
  }

  internal fun handlerLog(tkLog: TKLogModel) {
    ThreadPoolUtils.threadPool?.execute {
      destinations.forEach { destination ->
        destination.handlerLog(tkLog)
      }
    }
  }

}

File: ios/Classes/TKLoggerFlutter.swift
//
//  TKLoggerFlutter.swift
//  Pods-Runner
//
//  Created by Shper on 2020/7/24.
//

import Foundation
import TKLogger

public class TKLoggerFlutter {
    
    private static var destinations = Set<TKLogBaseDestination>()
    
    private init() {
        
    }
    
    // MARK: Destination
    
    @discardableResult
    public static func addDestination(_ destination: TKLogBaseDestination) -> Bool {
        if destinations.contains(destination) {
            return false
        }
        
        destinations.insert(destination)
        return true
    }
    
    // MARK: HandlerLog
    
    static func handlerLog(_ tkLog: TKLogModel) {
        DispatchQueue.global().async {
            destinations.forEach { destination in
                 destination.handlerLog(tkLog)
            }
        }
    }
    
}
